tests/unit/models/Component.spec.js

```diff
@@ -49,44 +51,213 @@ describe('Test class: Component', () => {
+        const component = new Component({
+          attributes: [
+            new ComponentAttribute({
+              name: 'test',
+              definition: new ComponentAttributeDefinition({
+                name: 'container',
+                type: 'test',
+                containerRef: 'container',
+              }),
+              value: 'containerId',
+            }),
+            new ComponentAttribute({
+              name: 'container2',
+              definition: new ComponentAttributeDefinition({
+                name: 'container2',
+                type: 'Reference',
+                containerRef: 'container2',
+              }),
+              value: 'containerId',
+            }),
+          ],
+        });
+        component.unsetContainerAttribute(new Component({
+          id: 'containerId',
+          definition: new ComponentDefinition({
+            type: 'container',
+          }),
+        }));
+
+        expect(component.attributes.length)
+          .toEqual(2);
+      });
+
+      it('Should do nothing if attribute not exists', () => {
+        const component = new Component({
+          attributes: [
+            new ComponentAttribute({
+              name: 'test',
+              definition: new ComponentAttributeDefinition({
+                name: 'container',
+                type: 'test',
+                containerRef: 'container',
+              }),
+              value: 'containerId',
+            }),
+            new ComponentAttribute({
+              name: 'container2',
+              definition: new ComponentAttributeDefinition({
+                name: 'container2',
+                type: 'Reference',
+                containerRef: 'container2',
+              }),
+              value: 'containerId',
+            }),
+          ],
+        });
+        component.unsetContainerAttribute(new Component({
+          id: 'containerId',
+          definition: new ComponentDefinition({
+            type: 'container',
+          }),
+        }));
+
+        expect(component.attributes.length)
+          .toEqual(2);
+      });
+    });
+
+    describe('Test method: unsetAllContainerAttribute', () => {
+      it('Should remove all container attribute', () => {
+        const component = new Component({
+          attributes: [
+            new ComponentAttribute({
+              name: 'container',
+              definition: new ComponentAttributeDefinition({
+                name: 'container',
+                type: 'Reference',
+                containerRef: 'container',
+              }),
+              value: 'containerId',
+            }),
+            new ComponentAttribute({
+              name: 'name',
+              definition: new ComponentAttributeDefinition({
+                name: 'container',
+                type: 'String',
+              }),
+              value: 'test',
+            }),
+          ],
+        });
+        component.unsetAllContainerAttribute();
+
```

Comment:
explicitly test that all the correct attributes have been removed

Reply:
done